### ISSUE TITLE
PyPi mix __version__ and platform if dash present.

### ISSUE THREAD

**Issue by Carreau** [takluyver/flit#34]
I guess this is a PyPI bug, but we should check for it, that version match pep 440, though I don't findknow where we can find a pep 440 version parser. 

with `__version__ = '0.0.4-dev0'`: `got requests.exceptions.HTTPError: 400 Client Error: Binary wheel for an unsupported platform`

```
$ flit wheel --upload
Created dist/SPQR-0.0.4-dev0-py3-none-any.whl                                                                                                           I-flit.wheel
Using repository at https://pypi.python.org/pypi                                                                                                       I-flit.upload
Uploading dist/SPQR-0.0.4-dev0-py3-none-any.whl...                                                                                                     I-flit.upload
Starting new HTTPS connection (1): pypi.python.org                                                                        I-requests.packages.urllib3.connectionpool
Traceback (most recent call last):
  File "/Users/bussonniermatthias/Library/Python/3.4/bin/flit", line 4, in <module>
    main()
  File "/Users/bussonniermatthias/Library/Python/3.4/lib/python/site-packages/flit/__init__.py", line 67, in main
    repo=args.repository).build()
  File "/Users/bussonniermatthias/Library/Python/3.4/lib/python/site-packages/flit/wheel.py", line 152, in build
    self.post_build()
  File "/Users/bussonniermatthias/Library/Python/3.4/lib/python/site-packages/flit/wheel.py", line 144, in post_build
    do_upload(self.wheel_file, self.metadata, self.repo)
  File "/Users/bussonniermatthias/Library/Python/3.4/lib/python/site-packages/flit/upload.py", line 164, in do_upload
    upload_wheel(file, metadata, repo)
  File "/Users/bussonniermatthias/Library/Python/3.4/lib/python/site-packages/flit/upload.py", line 133, in upload_wheel
    resp.raise_for_status()
  File "/usr/local/lib/python3.4/site-packages/requests/models.py", line 851, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Binary wheel for an unsupported platform
```

**Comment by takluyver** [comment]
If we want a simpler fix, I'd guess that just replacing `-` with `_` might work.

**Comment by Carreau** [comment]
or by `.`. I would still like a way to match the version string to pep 440.

**Comment by takluyver** [comment]
It should be simple enough to implement something based on the [normalisation rules](https://www.python.org/dev/peps/pep-0440/#normalization) in the PEP, right?

**Comment by Carreau** [comment]
> On Jun 1, 2015, at 10:09, Thomas Kluyver <notifications@github.com> wrote:
> 
> It should be simple enough to implement something based on the normalisation rules <https://www.python.org/dev/peps/pep-0440/#normalization> in the PEP, right?
> 
Sure but I was thinking this might have been implemented somewhere else.

**Comment by takluyver** [comment]
I think this should have been fixed by my recent changes for #53. We can reopen if there's still a problem.
